addons/website/static/src/snippets/s_share/000.js

```diff
@@ -75,10 +74,19 @@ const ShareWidget = publicWidget.Widget.extend({
                     return `${a + title}%20${url + c}`;
                 }
                 return a + title + c;
-            })
-            .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
             });
+        if (aEl.href.match(this.MEDIA_REGEX)) {
+            const ogImageEl = document.querySelector("meta[property='og:image']");
+            const url = new URL(aEl.href);
+            // Some pages (/profile/user/ID) don't have an image to share.
+            if (ogImageEl) {
+                const media = encodeURIComponent(ogImageEl.content);
+                url.searchParams.set("media", media);
+            } else {
+                url.searchParams.delete("media");
+            }
+            aEl.href = url.toString();
+        }
```

Comment:
@Guillaume-gdi The spirit of this code will be (arguably) better than the one-line code I suggested with the very simple regex but this code here in stable does not make sense: `MEDIA_REGEX` is a complex regex  which is there to capture stuff, ignore the capture of other stuff, etc... but you end up using it for testing the presence of "media=" only (you should actually use `test` instead of `match`). `MEDIA_REGEX` can probably be as simple as `/[&?]media=/` here... but as you cannot change the regex saved in `this` in stable, I would reserve doing this for master. Where in the end your `if` would be `if (url.searchParams.get('media') {` I guess (you would not need a regex at all).

So I suggest you do this:
- [Go back to the suggestion I made for stable](https://github.com/odoo/odoo/pull/133515#discussion_r1330321341)
- In the forward-ported master version, we can stop using the regex and use this :+1: Probably at the same time, not using it for title/url/etc and only manipulate the params directly with `searchParams` as you did.

Reply:
The [solution you suggested](https://github.com/odoo/odoo/pull/133515#discussion_r1330321341) doesn't work. It has to be adapted to something like this to work (not sure it works for every cases)
```js
if (a.startsWith("?") && c === "&") {
   return a.replace(/media=/, "");
}
return a.replace(/[&?]media=/, "") + c;
```

Why not stop using `this.MEDIA_REGEX` from this stable version and use `if (url.searchParams.has('media') {` instead. We can keep the definition of `this.MEDIA_REGEX` with a comment explaining why we're keeping it until the master version even if it is not used anymore.